### PR TITLE
[python] add placeholders to titles in plotting functions

### PR DESCRIPTION
**Pull request by StrikerRUS**
Similarly to `title` argument in `plot_split_value_histogram()` function:

https://github.com/microsoft/LightGBM/blob/d10f9d43c490d3e49afc4fdcb72a129cfaf70777/python-package/lightgbm/plotting.py#L199-L205

### CODE REVIEW

File: python-package/lightgbm/plotting.py [hunk 1]
```diff
@@ -370,14 +373,17 @@ def plot_metric(
         if metric not in metrics_for_one:
             raise KeyError('No given metric in eval results.')
         results = metrics_for_one[metric]
-    num_iteration, max_result, min_result = len(results), max(results), min(results)
+    num_iteration = len(results)
+    max_result = max(results)
+    min_result = min(results)
```
Comment (jameslamb) [inline python-package/lightgbm/plotting.py:378]: thank you for breaking this up!

File: tests/python_package_test/test_plotting.py [hunk 2]
```diff
@@ -68,6 +68,13 @@ def test_plot_importance(params, breast_cancer_split, train_data):
     assert ax2.patches[2].get_facecolor() == (0, .5, 0, 1.)  # g
     assert ax2.patches[3].get_facecolor() == (0, 0, 1., 1.)  # b
 
+    ax3 = lgb.plot_importance(gbm0, title='t @importance_type@', xlabel='x @importance_type@', ylabel='y @importance_type@')
+    assert isinstance(ax3, matplotlib.axes.Axes)
+    assert ax3.get_title() == 't @importance_type@'
```
Comment (jameslamb) [inline tests/python_package_test/test_plotting.py:73]: I'm confused by these tests. Shouldn't the template string `@importance_type@` have been replaced with the actual value of `importance_type`?

**Review by jameslamb — COMMENTED** [review]
I support this change but I don't understand the tests, left a question about it.

File: tests/python_package_test/test_plotting.py (hunk 2 above)
Comment (StrikerRUS) [reply tests/python_package_test/test_plotting.py:73]: Only in `xlabel` argument, not in `title` or any other.
https://github.com/microsoft/LightGBM/blob/6714a0f7fcfb2f5f0ea1adfa4c45b4ef0501098c/python-package/lightgbm/plotting.py#L62-L65

Comment (jameslamb) [reply tests/python_package_test/test_plotting.py:73]: ah I see, think I was confused by changes to the multiple different functions. Thanks for the explanation.

**Review by jameslamb — APPROVED** [review]
very nice addition, thanks!

**Pull request merged**